cleanlab_studio/utils/data_enrichment/enrich.py

```diff
@@ -94,41 +74,78 @@ def enrich_data(
     df[f"{new_column_name}"] = [
         output["response"] if output is not None else None for output in outputs
     ]
-
-    if (
-        regex is None and constrain_outputs is None
-    ):  # we do not need to have a "log" column as original output is not augmented by regex replacements or contrained outputs
-        return df[[f"{new_column_name}", f"{column_name_prefix}trustworthiness"]]
-
     df[f"{column_name_prefix}log"] = [
         output["response"] if output is not None else None for output in outputs
     ]
 
-    if regex:
-        df[f"{new_column_name}"] = df[f"{new_column_name}"].apply(
-            lambda x: get_regex_match_or_replacement(x, regex)
-        )
+    if regex is None and constrain_outputs is None:
+        enriched_df = df[[f"{new_column_name}", f"{column_name_prefix}trustworthiness"]]
+    else:
+        if regex:
+            df[f"{new_column_name}"] = df[f"{new_column_name}"].apply(
+                lambda x: get_regex_match_or_replacement(x, regex)
+            )
 
-    if constrain_outputs:
-        df[f"{new_column_name}"] = df[f"{new_column_name}"].apply(
-            lambda x: get_constrain_outputs_match(
-                x, constrain_outputs, disable_warnings=disable_warnings
+        if constrain_outputs:
+            df[f"{new_column_name}"] = df[f"{new_column_name}"].apply(
+                lambda x: get_constrain_outputs_match(x, constrain_outputs)
             )
-        )
 
-    return df[
-        [
-            f"{new_column_name}",
-            f"{column_name_prefix}trustworthiness",
-            f"{column_name_prefix}log",
+        enriched_df = df[
+            [
+                f"{new_column_name}",
+                f"{column_name_prefix}trustworthiness",
+                f"{column_name_prefix}log",
+            ]
         ]
-    ]
+
+    # Simulate the response structure of the run() method
+    job_info = {
+        "job_id": "run_online",
+        "status": "SUCCEEDED",
+        "num_rows": len(enriched_df),
+        "processed_rows": len(enriched_df),
+        "average_trustworthiness_score": enriched_df[f"{column_name_prefix}trustworthiness"].mean(),
+        "results": enriched_df,
+    }
+
+    return job_info
+
+
+def _validate_enrichment_options(options: EnrichmentOptions) -> None:
```

Comment:
Can you clarify why there is a separate `_validate_enrichment_options` defined here rather than using the validation function in `run()` [here](https://github.com/cleanlab/cleanlab-studio/blob/refactor-dynamic-static-enrichment-apis/cleanlab_studio/studio/enrichment.py#L190)?